src/engine/renderer/tr_image.cpp

```diff
@@ -1937,23 +1937,53 @@ static void R_Rotate( byte *in, int width, int height, int degrees )
 	ri.Hunk_FreeTempMemory( tmp );
 }
 
+byte *R_Resize( byte *in, int width, int height, int width2, int height2 )
```

Comment:
This could use a comment or a different name to indicate that it's specifically for upsampling, not any resize.

Reply:
do you think that algorithm does not work for downscaling?
we only use it to upscale, but isn't it generic enough to downscale too?

Reply:
Well maybe, if not with the best quality. But there's already a downsampling function `ResampleTexture` in the same file.